src/dawn/CodeGen/ASTCodeGenGTClangStencilDesc.cpp

```diff
@@ -20,6 +20,31 @@
 
 namespace dawn {
 
+static Extents analyzestencilExtents(const std::shared_ptr<Stencil>& s, int ID) {
```

Comment:
Can we have some doc describing the alrgorithm, i.e. what is suppose to return? 
Also the parameters, ID is not clear whether is the ID of a field, stencil, etc... so I would also probably use a longer name